modules/api/src/main/java/com/intuit/wasabi/api/AssignmentsResource.java

```diff
@@ -645,11 +622,94 @@ public Response clearMetadataCache(@HeaderParam(AUTHORIZATION) @ApiParam(value =
-        UserInfo.Username userName = authorization.getUser(authorizationHeader);
-        authorization.checkSuperAdmin(userName);
-
+    public Response getMetadataCacheDetails() {
         return httpHeader.headers().entity(assignments.metadataCacheDetails()).build();
     }
 
+    /**
+     * Convert Assignment object collection to the List of response MAP expected by the end user.
+     *
+     * @param assignments
+     * @return
+     */
+    protected List<Map<String, Object>> toBatchAssignmentResponseMap(Collection<Assignment> assignments) {
+        List<Map<String, Object>> responseList = new ArrayList<>();
+        assignments.forEach(assignment -> {
+            responseList.add(toBatchAssignmentResponseMap(assignment));
+        });
+        return responseList;
+    }
+
+    /**
+     * Convert Assignment object of batch-assignment to the response MAP expected by the end user.
+     * Batch-assignment response map can contain:
+     *  - experimentLabel
+     *  - assignment
+     *  - payload
+     *  - status
+     *
+     * @param assignment
+     *
+     * @return response map
+     *
+     */
+    protected Map<String, Object> toBatchAssignmentResponseMap(final Assignment assignment) {
+        Map<String, Object> response = newHashMap();
+
+        //Add experimentLabel for batch-assignment flow only
+        if(nonNull(assignment.getExperimentLabel())) {
+            response.put("experimentLabel", assignment.getExperimentLabel());
+        }
+
+        // Only include `assignment` property if there is a definitive assignment, either to a bucket or not
+        if (assignment.getStatus() != EXPERIMENT_EXPIRED) {
+            response.put("assignment", nonNull(assignment.getBucketLabel()) ? assignment.getBucketLabel().toString() : null);
+
+            if (nonNull(assignment.getBucketLabel())) {
+                response.put("payload", assignment.getPayload());
+            }
+        }
+
+        response.put("status", assignment.getStatus());
+
+        return response;
+    }
+
+    /**
+     * Convert Assignment object of single-assignment to the response MAP expected by the end user.
+     *
+     *  Single-assignment response map can contain:
+     *  - assignment
+     *  - payload
+     *  - status
+     *  - cache
+     *  - context
+     *
+     * @param assignment
+     *
+     * @return response map
+     *
+     */
+    protected Map<String, Object> toSingleAssignmentResponseMap(final Assignment assignment) {
+        Map<String, Object> response = newHashMap();
+
+        // Only include `assignment` property if there is a definitive assignment, either to a bucket or not
+        if (assignment.getStatus().isDefinitiveAssignment()) {
+            response.put("assignment", nonNull(assignment.getBucketLabel()) ? assignment.getBucketLabel().toString() : null);
+
+            if (nonNull(assignment.getBucketLabel())) {
```

Comment:
it is not, because in case of definitive-assignment, even if bucket label is NULL then we still want to include 'assignment' key in the response map but not the 'payload' key...